### ISSUE TITLE
AllowUnJoined issue

### ISSUE THREAD

**Issue by x1p** [Ben12345rocks/VotingPlugin#1208]
**Versions**
VotingPlugin v 6.7 (stable)
Server version 1.17.1

**Describe the bug**
A player made an error filling in his name on a vote site. Instead of his name, he only filled in just the first letter: `F`
Votesite accepted it, and the vote arrived on the server.

The bungee contains `AllowUnJoined: false` The player `F` apparently indeed exists according to namemc.com.
This player never joined our server. So it's non-existant according to our databases.

Nevertheless VotingPlugin accepted it.

**To Reproduce**
1. Setup a network using VotinPlugin on the bungee and on the servers attached to the bungee
2. Vote for player "F"
3. See console
4. See broadcast in game

**Expected behavior**
A player that has never joined our server should not be broadcasted.

**Screenshots/Configs**
Bungeecord config setup:
`AllowUnJoined: false`
Paper config setup:
`AllowUnjoined: true`
(according to the bungeeconfig on the bungeecord, the servers should allow unjoined if the bungee filters this)

**Additional context**
Bungee log:
[21:23:08] [Votifier epoll worker/INFO] [NuVotifier]: Got a protocol v1 vote record from /ipaddressredacted:11111 -> Vote (from:votesiteredacted username:F address:ipaddressredacted timeStamp:1638130988 additionalData:null)
[21:23:08] [VotingPlugin Pool Thread 272/INFO] [VotingPlugin]: Vote received F from service site votesiteredacted

Server log:
[21:34:16] [Craft Scheduler Thread - 813 - VotingPlugin/INFO]: [VotingPlugin] BungeeDebug: Received plugin message: Vote, F, 3a61c62b-0d9f-406f-ae63-8e6a0fe5a284, votesiteredacted, 1638130988710, false, true, 3//1//1//1//3//3, true, 1, true

**Comment by BenCodez** [comment]
I just did a test and it works for me, got an easy way to recreate it?

**Comment by x1p** [comment]
I'm afraid not. Do you have any idea?
I suppose I can set the individual servers to `AllowUnJoined: false` as a fallback but I'm not sure why it's not advised.

Edit: I searched multiple databases. That uuid it unknown to me.

**Comment by BenCodez** [comment]
Don't even think that would really work, as the vote is already partly processed by that point, I think it would still broadcast

**Comment by BenCodez** [comment]
https://bencodez.com/job/VotingPlugin/ I added more debug, in order to determine why it thinks it's a valid player, see what happens? https://bencodez.com/job/VotingPlugin/ applies to bungee only for what is needed to debug this, make sure to enable debug

**Comment by x1p** [comment]
I thouhgt I'd try `/votepoints f` and it turns out this 'player" already has 3 points.
So this entry is apparently already in the votingdatabase. I allways had this AllowUnJoined set to false so I'm not sure when this got entry to slip in.
So I guess I just delete this from the database and it should be solved?

**Comment by BenCodez** [comment]
should be yes

**Comment by x1p** [comment]
I checked the database for more names and there were another 2 names with single characters. Quite a coincidence.
I'll check tomorrow to see if it's still accepted.

**Comment by BenCodez** [comment]
Hmm, interesting, not sure why it's like that

**Comment by x1p** [comment]
Tested it again today.
`[VotingPlugin]: Ignoring vote from F since player hasn't joined before`

I also manually purged all old players (with 0 points who bought nothing) from the database to minimize the change of weird names comming through. This was half the database and several thousands entries of useless data.

So I guess this issue can be closed although I'm not exactly sure how the F and X entries came in the database in the first place.
